community/modules/scripts/spack-install/variables.tf

```diff
@@ -92,138 +68,126 @@ variable "compilers" {
   type        = list(string)
 }
 
-variable "licenses" {
-  description = "List of software licenses to install within spack."
-  default     = null
-  type = list(object({
-    source = string
-    dest   = string
-  }))
-}
 
 variable "packages" {
   description = "Defines root packages for spack to install (in order)."
   default     = []
   type        = list(string)
 }
 
-variable "install_flags" {
-  description = "Defines the flags to pass into `spack install`"
-  default     = ""
-  type        = string
-}
-
-variable "concretize_flags" {
-  description = "Defines the flags to pass into `spack concretize`"
-  default     = ""
-  type        = string
-}
+/* Deprecated Functionality */
 
-variable "gpg_keys" {
-  description = <<EOT
-  GPG Keys to trust within spack.
-  Each key must define a type. Valid types are 'file' and 'new'.
-  Keys of type 'file' must define a path to the key that
-  should be trusted.
-  Keys of type 'new' must define a 'name' and 'email' to create
-  the key with.
-EOT
-  default     = []
-  type        = list(map(any))
-  validation {
-    condition = alltrue([
-      for k in var.gpg_keys : contains(keys(k), "type")
-    ])
-    error_message = "Each gpg_key must define a type."
-  }
-  validation {
-    condition = alltrue([
-      for k in var.gpg_keys : (k["type"] == "file" || k["type"] == "new")
-    ])
-    error_message = "Valid types for gpg_keys are 'file' and 'new'."
-  }
+variable "spack_cache_url" {
+  description = "DEPRECATED"
+  type        = any
+  default     = null
+  nullable    = true
   validation {
-    condition = alltrue([
-      for k in var.gpg_keys : ((k["type"] == "file" && contains(keys(k), "path")) || (k["type"] == "new"))
-    ])
-    error_message = "Each gpg_key of type file must define a path."
+    condition     = var.spack_cache_url == null
+    error_message = <<EOT
+    The spack_cache_url setting is deprecated.
```

Comment:
nit: consistent indentation. Either use `<<-EOT` with the `EOT` in line with the text or remove the indentation. Same goes for the other blocks.

Reply:
Fixed, this and the others. Terraform was not accepting this syntax actually, so it had to be updated.